micropip/_micropip.py

```diff
@@ -692,3 +698,170 @@ def _list():
             source_ = pkg_source
         packages[name] = PackageMetadata(name=name, version=version, source=source_)
     return packages
+
+
+MOCK_INSTALL_NAME_MEMORY = "micropip in-memory mock package"
+MOCK_INSTALL_NAME_PERSISTENT = "micropip mock package"
+
+
+def add_mock_package(
+    name: str, version: str, *, modules: dict | None = None, persistent: bool = False
+) -> None:
+    """
+       Add a mock version of a package to the package dictionary.
+
+       This means that if it is a dependency, it is skipped on install.
+
+       By default a single empty module is installed with the same
+       name as the package. You can alternatively give one or more modules to make a
+       set of named modules.
+
+       The modules parameter is usually a dictionary mapping module name to module text.
+
+        e.g.
+       ``
+       {
+          "mylovely_module":'''
+            def module_method(an_argument):
+                print("This becomes a module level argument")
+            module_value = "this value becomes a module level variable"
+
+            print("This is run on import of module")
+          '''
+       }
+
+       If you are adding the module in non-persistent mode, you can also pass functions
+       which are used to initialize the module on loading (as in `importlib.abc.loader.exec_module` ).
+       This allows you to do things like use `unittest.mock.MagicMock` classes for modules.
+
+        e.g.
+       ``
+       def init_fn(module):
+           module.dict["WOO"]="hello"
+           print("Initing the module now!")
+
+       ...
+       ...
+
+       {
+          "mylovely_module": init_fn
+       }
+       ``
+
+    Parameters
```

Comment:
I think there is an indentation issue in this docstring, but we can fix it once we start actually building docs for it.